pkg/provisioner/ironic/ironic.go

```diff
@@ -1368,3 +1371,107 @@ func (p *ironicProvisioner) softPowerOff() (result provisioner.Result, err error
 
 	return result, nil
 }
+
+// Describes the current availability status of a generic dependency
+// required by the provisioner to operate correctly
+type provisionerDependencyStatus struct {
+	// The name of the dependency
+	name string
+	// If ready, the dependency is available to process incoming requests
+	ready bool
+	// Can provide additional info if the dependency is not ready
+	err error
+}
+
+// Check to verify dependency status
+type provisionerDependencyCheck func(chan<- provisionerDependencyStatus)
+
+// IsReady checks if the provisioning backend is available
+func (p *ironicProvisioner) IsReady() (bool, error) {
+	p.log.Info("verifying ironic provisioner dependencies")
+
+	checks := []provisionerDependencyCheck{
+		p.checkIronic,
+		p.checkInspector,
+	}
+
+	var done = make(chan provisionerDependencyStatus, len(checks))
+	defer close(done)
+
+	for _, check := range checks {
+		go check(done)
+	}
+
+	results := []provisionerDependencyStatus{}
+	for i := 0; i < len(checks); i++ {
+		results = append(results, <-done)
+	}
+
+	for _, r := range results {
+		if !r.ready {
+			p.log.Info("dependency not yet ready", "name", r.name)
+			return r.ready, r.err
+		}
+	}
+
+	return true, nil
+}
+
+func (p *ironicProvisioner) checkIronic(done chan<- provisionerDependencyStatus) {
```

Comment:
It might be cleaner to put all of this new logic into a new type in a separate file in this package, and not have it be methods of the Provisioner other than the new IsReady() function required by the interface.

Reply:
Right - I did not want to overengineer the first proposal

Reply:
While working on it, I found it convenient to split in two, a generic part in the `provisionerdependencies.go` file and a specific Ironic one in the `ironicdependecies.go` file